File: app/src/main/java/com/asafvaron/themoviedbtest/ui/mvp_favorites/FavoritesContract.java
package com.asafvaron.themoviedbtest.ui.mvp_favorites;

import com.asafvaron.themoviedbtest.model.Movie;

import java.util.List;

/**
 * Created by asafvaron on 05/03/2017.
 */
interface FavoritesContract {
    interface View {
        void onLoadComplete(List<Movie> favsList);

        void updateProgress(boolean isUpdating);
    }

    interface UserActions {

        void loadFavorites();
    }
}

File: app/src/main/java/com/asafvaron/themoviedbtest/ui/mvp_favorites/FavoritesContract.kt
package com.asafvaron.themoviedbtest.ui.mvp_favorites

import com.asafvaron.themoviedbtest.model.Movie

/**
 * Created by asafvaron on 05/03/2017.
 */
internal interface FavoritesContract {
    interface View {
        fun onLoadComplete(favsList: List<Movie>? = null)

        fun updateProgress(isUpdating: Boolean)
    }

    interface Presenter {

        fun loadFavorites()
    }
}
